serializer.rst

```diff
@@ -221,310 +189,250 @@ which is creating a PHP object from a formatted string (e.g. JSON or XML):
+    when updating data). See ...
 
-* JSON: :class:`Symfony\\Component\\Serializer\\Encoder\\JsonEncoder`
-* XML: :class:`Symfony\\Component\\Serializer\\Encoder\\XmlEncoder`
-* CSV: :class:`Symfony\\Component\\Serializer\\Encoder\\CsvEncoder`
-* YAML: :class:`Symfony\\Component\\Serializer\\Encoder\\YamlEncoder`
+The Serialization Process: Normalizers and Encoders
+---------------------------------------------------
 
-As well as the following normalizers:
+The serializer uses a two-step process when (de)serializing objects:
 
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\ObjectNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\DateTimeNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\DateTimeZoneNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\DateIntervalNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\FormErrorNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\DataUriNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\JsonSerializableNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\ArrayDenormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\ConstraintViolationListNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\ProblemNormalizer`
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\BackedEnumNormalizer`
-
-.. versionadded:: 5.4
+.. raw:: html
 
-    :class:`Symfony\\Component\\Serializer\\Normalizer\\BackedEnumNormalizer`
-    was introduced in Symfony 5.4. PHP BackedEnum requires at least PHP 8.1.
+    <object data="../_images/components/serializer/serializer_workflow.svg" type="image/svg+xml"
+        alt="A diagram with a block representing the object at the top, and
+        one representing the format (JSON, XML, CSV) at the bottom. An
+        arrow pointing from the format to the object is labelled
+        'deserialize', while an opposite arrow is labelled 'serialize'. In
+        between the blocks, a block named 'array' exists. Arrows in between
+        the object and array in the serialize direction are called
+        'normalize' and the opposite 'denormalize', while arrows in between
+        the array and the format are called 'encode' in serialize
+        direction, and 'decode' when deserializing."
+    ></object>
+
+In both directions, data is always first converted to an array. This splits
+the process in two seperate responsiblities:
+
+Normalizers
+    These classes convert **objects** into **arrays** and vice versa. They
+    do the heavy lifting of finding out which class properties to
+    serialize, what value they hold and what name they should have.
+Encoders
+    Encoders convert **arrays** into a specific **format** and the other
+    way around. Each encoder knows exactly how to parse and generate a
+    specific format, for instance JSON or XML.
+
+Internally, the ``Serializer`` class uses a sorted list of normalizers and
```

Comment:
"a sorted list" ... could we mention how is that list sorted? (by priority? by registration order? other?) Maybe it's something explained later or not important to document.

Reply:
Yeah, this is TBD still. It's "first supported normalizer takes it all", so I thought it is important enough to mention - but it'll be only important how this is done exactly once you implement a custom normalizer (which I plan to explain at the end of this article).